backend-api/src/functions/adapters/dynamoDbAdapter.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  AttributeValue,
+  ConditionalCheckFailedException,
+  DynamoDBClient,
+  PutItemCommand,
+  PutItemCommandInput,
+  QueryCommand,
+  QueryCommandInput,
+  QueryCommandOutput,
+} from "@aws-sdk/client-dynamodb";
+import { errorResult, Result, successResult } from "../utils/result";
+import {
+  CreateSessionAttributes,
+  SessionDetails,
+} from "../services/sessionService/sessionService";
+import { NodeHttpHandler } from "@smithy/node-http-handler";
+
+type DynamoDbRecord = Record<string, AttributeValue>;
+
+export class DynamoDbAdapter {
+  private readonly tableName: string;
+  private readonly dynamoDbClient = new DynamoDBClient({
+    region: process.env.REGION,
+    maxAttempts: 2,
+    requestHandler: new NodeHttpHandler({
+      connectionTimeout: 29000,
+      requestTimeout: 29000,
+    }),
+  });
+
+  constructor(tableName: string) {
+    this.tableName = tableName;
+  }
+
+  async readSessionId(
+    subjectIdentifier: string,
+  ): Promise<Result<string | null>> {
```

Comment:
in which instance could this be null? I might be missing something

Reply:
When there is no active session

Reply:
Thank you. I misread this block:

`if (!sessionId) {
      return errorResult({
        errorMessage: "Session is malformed",
        errorCategory: "SERVER_ERROR",
      });
    }`